x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/shared/map_processor_type_to_form.tsx

```diff
@@ -192,46 +286,84 @@ export const mapProcessorTypeToDescriptor: MapProcessorTypeToDescriptor = {
         />
       );
     },
+    getDefaultDescription: ({ field, policy_name: policyName, target_field: targetField }) =>
+      i18n.translate('xpack.ingestPipelines.processors.defaultDescription.enrich', {
+        defaultMessage:
+          'Enriches data to "{target_field}" if the "{policy_name}" policy matches "{field}"',
+        values: {
+          field,
+          policy_name: policyName,
+          target_field: targetField,
+        },
+      }),
   },
   fail: {
     FieldsComponent: Fail,
     docLinkPath: '/fail-processor.html',
     label: i18n.translate('xpack.ingestPipelines.processors.label.fail', {
       defaultMessage: 'Fail',
     }),
-    description: i18n.translate('xpack.ingestPipelines.processors.description.fail', {
+    typeDescription: i18n.translate('xpack.ingestPipelines.processors.description.fail', {
       defaultMessage:
         'Returns a custom error message on failure. Often used to notify requesters of required conditions.',
     }),
+    getDefaultDescription: ({ if: value }) =>
+      value
+        ? i18n.translate('xpack.ingestPipelines.processors.defaultDescription.fail', {
+            defaultMessage: 'Raises an exception if the current conditions are met "{if}"',
+            values: {
+              if: value,
+            },
+          })
+        : undefined,
   },
   foreach: {
     FieldsComponent: Foreach,
     docLinkPath: '/foreach-processor.html',
     label: i18n.translate('xpack.ingestPipelines.processors.label.foreach', {
       defaultMessage: 'Foreach',
     }),
-    description: i18n.translate('xpack.ingestPipelines.processors.description.foreach', {
+    typeDescription: i18n.translate('xpack.ingestPipelines.processors.description.foreach', {
       defaultMessage: 'Applies an ingest processor to each value in an array.',
     }),
+    getDefaultDescription: ({ field, processor }) => {
+      const processorName = Object.keys(processor ?? {})[0];
+      return processorName
+        ? i18n.translate('xpack.ingestPipelines.processors.defaultDescription.foreach', {
```

Comment:
```suggestion
        i18n.translate('xpack.ingestPipelines.processors.defaultDescription.foreach', {
```

Reply:
The description for Drop wasn't rendering, and I think the errant `?` might be why.

Reply:
@lockewritesdocs This is the description for the Foreach processor. The `?` guards us from trying to build a default description if we bad data. This is a limitation that will be removed once we address https://github.com/elastic/kibana/issues/95906. Let me know if this is unclear!